log/logtest/recorder.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package logtest is a testing helper package. User can retrieve an in-memory
+// logger to verify the behavior of their integrations.
+package logtest // import "go.opentelemetry.io/otel/log/logtest"
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/embedded"
+)
+
+// embeddedLogger is a type alias so the embedded.Logger type doesn't conflict
+// with the Logger method of the recorder when it is embedded.
+type embeddedLogger = embedded.Logger // nolint:unused  // Used below.
+
+type enabledFn func(context.Context, log.Record) bool
+
+var defaultEnabledFunc = func(context.Context, log.Record) bool {
+	return true
+}
+
+type config struct {
+	enabledFn enabledFn
+}
+
+func newConfig(options []Option) config {
+	var c config
+	for _, opt := range options {
+		c = opt.apply(c)
+	}
+
+	return c
+}
+
+// Option configures a [Recorder].
+type Option interface {
+	apply(config) config
+}
+
+type optFunc func(config) config
+
+func (f optFunc) apply(c config) config { return f(c) }
+
+// WithEnabledFunc allows configuring whether the recorder enables specific log entries or not.
+//
+// By default, every log entry will be enabled.
+func WithEnabledFunc(fn enabledFn) Option {
```

Comment:
`enabledFn` needs to be exported, or a named type should not be used here.

```suggestion
func WithEnabledFunc(fn func(context.Context, log.Record) bool) Option {
```